pkg/provider/azure_loadbalancer.go

```diff
@@ -801,6 +801,12 @@ func (az *Cloud) ensurePublicIPExists(service *v1.Service, pipName string, domai
 			pip.Sku = &network.PublicIPAddressSku{
 				Name: network.PublicIPAddressSkuNameStandard,
 			}
+
+			// only add zone information for the new standard pips
+			zones := az.regionZonesMap[to.String(pip.Location)]
```

Comment:
need to wait for zonesmap initialized on startup. Could you add another wrapper function which would do:

1) Check whether zonesmap cached
2) if not, invoke zone API to initiliaze cache
3) return the supported zones in specified location

since regionZonesMap would be accessed in multi go routines, a lock should also be added

Reply:
added a config `refreshZonesRetryCount` and invoke zone api with retry before put pip/lb if the zones map is empty